cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/distributed/DistributedProgramRunner.java

```
@@ -801,4 +807,27 @@ private Set<Class<?>> addExtraDependencies(CConfiguration cConf, Set<Class<?>> d
 
     return dependencies;
   }
+
+  @VisibleForTesting
+  static void copyRuntimeToken(LocationFactory locationFactory, Arguments args, File tempDir,
+                               Map<String, LocalizeResource> localizeResources) throws IOException {
+    File runtimeTokenFile = File.createTempFile("runtime-tether", ".token", tempDir);
+    Location location = locationFactory.create(URI.create(args.getOption(ProgramOptionConstants.PROGRAM_RESOURCE_URI)))
+      .append(Constants.Security.Authentication.RUNTIME_TOKEN_FILE);
+    if (!location.exists()) {
+      return;
+    }
+    boolean runtimeTokenCopied = true;
+    try (InputStream is = location.getInputStream()) {
+      Files.copy(is, runtimeTokenFile.toPath(), StandardCopyOption.REPLACE_EXISTING);
+      LOG.debug("Copied {} to {}", Constants.Security.Authentication.RUNTIME_TOKEN_FILE, runtimeTokenFile);
+    } catch (IOException e) {
+      LOG.warn("Failed to copy runtime token with exception, skipping...", e);
```

since the location exists, seems like an exception here should now cause the program run to fail (let the exception propagate up)

---

Fixed